mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td

```diff
@@ -706,6 +706,7 @@ def MemRef_DmaStartOp : MemRef_Op<"dma_start"> {
   let extraClassDeclaration = [{
     // Returns the source MemRefType for this DMA operation.
     Value getSrcMemRef() { return getOperand(0); }
+    OpOperand &getSrcMemRefMutable() { return getOperation()->getOpOperand(0); }
```

Comment:
I think this should not be necessary. `get...Mutable()` functions are auto-generated from Tablegen.

Reply:
srcMemRef is not a arg in td file,actually, the td file of DmaStartOp argument is :
```
    let arguments = (ins Variadic<AnyType>:$operands);
```

That also why we needs function getSrcMemRef in extraClassDeclaration.